easycord/_context_base.py

```
@@ -221,3 +221,26 @@ def get_member(self, user_id: int) -> discord.Member | None:
         """
         return self.guild.get_member(user_id) if self.guild else None
+
+    async def fetch_member(self, user_id: int) -> discord.Member:
+        """Fetch a guild member by user ID.
+
+        Tries the guild cache first; falls back to an API call.
+        Raises ``RuntimeError`` if called outside a guild.
+        Raises ``discord.NotFound`` if the user is not in the guild.
+        """
+        if self.guild is None:
+            raise RuntimeError("fetch_member requires a guild context")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)